Source/Core/Core/IOS/USB/Bluetooth/BTReal.cpp

```diff
@@ -398,167 +373,100 @@ void BluetoothRealDevice::TriggerSyncButtonHeldEvent()
   m_sync_button_state = SyncButtonState::LongPressed;
 }
 
-void BluetoothRealDevice::WaitForHCICommandComplete(const u16 opcode)
-{
-  int actual_length;
-  SHCIEventCommand packet;
-  std::vector<u8> buffer(1024);
-  // Only try 100 transfers at most, to avoid being stuck in an infinite loop
-  for (int tries = 0; tries < 100; ++tries)
-  {
-    const int ret = libusb_interrupt_transfer(m_handle, HCI_EVENT, buffer.data(),
-                                              static_cast<int>(buffer.size()), &actual_length, 20);
-    if (ret != LIBUSB_SUCCESS || actual_length < static_cast<int>(sizeof(packet)))
-      continue;
-    std::memcpy(&packet, buffer.data(), sizeof(packet));
-    if (packet.EventType == HCI_EVENT_COMMAND_COMPL && packet.Opcode == opcode)
-      break;
-  }
-}
-
 void BluetoothRealDevice::SendHCIResetCommand()
 {
-  u8 packet[3] = {};
-  const u16 payload[] = {HCI_CMD_RESET};
-  memcpy(packet, payload, sizeof(payload));
-  const int ret =
-      libusb_control_transfer(m_handle, REQUEST_TYPE, 0, 0, 0, packet, sizeof(packet), TIMEOUT);
-  if (ret < LIBUSB_SUCCESS)
-    WARN_LOG_FMT(IOS_WIIMOTE, "libusb_control_transfer failed: {}", LibusbUtils::ErrorWrap(ret));
-  else
-    INFO_LOG_FMT(IOS_WIIMOTE, "Sent a reset command to adapter");
+  INFO_LOG_FMT(IOS_WIIMOTE, "SendHCIResetCommand");
+  m_lib_usb_bt_adapter->SendControlTransfer(AsU8Span(hci_cmd_hdr_t{HCI_CMD_RESET, 0}));
 }
 
 void BluetoothRealDevice::SendHCIDeleteLinkKeyCommand()
 {
-  struct Payload
-  {
-    hci_cmd_hdr_t header;
-    hci_delete_stored_link_key_cp command;
-  };
-  Payload payload;
-  payload.header.opcode = HCI_CMD_DELETE_STORED_LINK_KEY;
-  payload.header.length = sizeof(payload.command);
+  INFO_LOG_FMT(IOS_WIIMOTE, "SendHCIDeleteLinkKeyCommand");
+
+  HCICommandPayload<HCI_CMD_DELETE_STORED_LINK_KEY, hci_delete_stored_link_key_cp> payload;
   payload.command.bdaddr = {};
-  payload.command.delete_all = true;
+  payload.command.delete_all = 0x01;
```

Comment:
Any particular reason this changed?

Reply:
It's a Bluetooth spec structure. The value is supposed to be 0x01 to delete all the keys https://www.bluetooth.com/wp-content/uploads/Files/Specification/HTML/Core-54/out/en/host-controller-interface/host-controller-interface-functional-specification.html#UUID-f81b6bd8-b4c1-38c5-ebf4-6a87e8e2da0e. Using `true` is weirdly relying on implicit conversions.

Reply:
Ah, I missed that the type is a uint.  Definitely agree with your comment then.